ingestion/src/test/java/feast/ingestion/ImportJobTest.java

```diff
@@ -180,91 +173,320 @@ public void runPipeline_ShouldWriteToRedisCorrectlyGivenValidSpecAndFeatureRow()
+      throws IOException, InterruptedException {
+
+    FeatureSetSpec rowSpec =
+        spec.toBuilder()
+            .clone()
+            .clearFeatures()
+            .addFeatures(
+                FeatureSpec.newBuilder()
+                    .setName("feature_1")
+                    .setValueType(Enum.STRING_LIST)
+                    .build())
+            .build();
+
+    List<FeatureRow> input = new ArrayList<>();
+
+    LOGGER.info("Generating test data ...");
+    IntStream.range(0, IMPORT_JOB_SAMPLE_FEATURE_ROW_SIZE)
+        .forEach(
+            i -> {
+              FeatureRow randomRow = TestUtil.createRandomFeatureRow(rowSpec);
+              input.add(randomRow);
+            });
+
+    Map<RedisKey, FeatureRow> expected = getExpectedRows(input, spec);
+    PipelineResult pipelineResult = startImportJob(options);
+    Assert.assertEquals(pipelineResult.getState(), State.RUNNING);
+    publishFeatureRows(input, pipelineResult);
 
     LOGGER.info("Validating the actual values written to Redis ...");
-    RedisClient redisClient =
-        RedisClient.create(new RedisURI(REDIS_HOST, REDIS_PORT, java.time.Duration.ofMillis(2000)));
     StatefulRedisConnection<byte[], byte[]> connection = redisClient.connect(new ByteArrayCodec());
     RedisCommands<byte[], byte[]> sync = connection.sync();
     expected.forEach(
         (key, expectedValue) -> {
 
           // Ensure ingested key exists.
-          byte[] actualByteValue = sync.get(key.toByteArray());
+          byte[] actualByteValue = getBytesFromKey(sync, key);
           if (actualByteValue == null) {
-            LOGGER.error("Key not found in Redis: " + key);
-            LOGGER.info("Redis INFO:");
-            LOGGER.info(sync.info());
-            byte[] randomKey = sync.randomkey();
-            if (randomKey != null) {
-              LOGGER.info("Sample random key, value (for debugging purpose):");
-              LOGGER.info("Key: " + randomKey);
-              LOGGER.info("Value: " + sync.get(randomKey));
-            }
             Assert.fail("Missing key in Redis.");
           }
 
           // Ensure value is a valid serialized FeatureRow object.
-          FeatureRow actualValue = null;
-          try {
-            actualValue = FeatureRow.parseFrom(actualByteValue);
-          } catch (InvalidProtocolBufferException e) {
-            Assert.fail(
-                String.format(
-                    "Actual Redis value cannot be parsed as FeatureRow, key: %s, value :%s",
-                    key, new String(actualByteValue, StandardCharsets.UTF_8)));
+          FeatureRow actualValue = getFeatureRowFromByte(key, actualByteValue);
+
+          // Ensure the retrieved FeatureRow is equal to the ingested FeatureRow.
+          Assert.assertEquals(expectedValue, actualValue);
+        });
+    redisClient.shutdown();
+  }
+
+  @Test
+  public void runPipeline_ShouldWriteFeatureRowsWithMissingEntityFieldsToDeadLetterQueue()
```

Comment:
Would you mind adding a few comments throughout this test to make it clearer what you are doing? It's a little bit cryptic as is.

Reply:
Same for your other test.

Reply:
Can we also update this thes method name?
Because it is NOT asserting that the invalid rows are written to a deadletter queue, it is only asserting that the rows with missing entity are not ingested.

Either that or I think we should update the test implemention to actually check that the ingestion job is writing to a deadletter queue.